worker/uniter/runner/runner.go

```diff
@@ -549,10 +596,27 @@ func (runner *runner) runCharmHookOnLocal(hookName string, env []string, charmLo
 			Stderr: readBytes(actionErr),
 		}
 		if err := runner.updateActionResults(resp); err != nil {
-			return errors.Trace(err)
+			return hookHandlerType, errors.Trace(err)
 		}
 	}
-	return errors.Trace(exitErr)
+
+	return hookHandlerType, errors.Trace(exitErr)
+}
+
+// getHook checks to see if the dispatch hook exists, if not, check for the
+// given hookName.
+func (runner *runner) getHook(hookName, charmDir, charmLocation string) (HookHandlerType, string, error) {
+	hook, err := searchHook(charmDir, hookDispatcherScript)
```

Comment:
LIkewise, `searchHook` is a bit misleading as to its purpose.

Reply:
renamed to discoverHookScript.